OData/src/System.Web.OData/OData/Builder/CapabilitiesVocabularyExtensionMethods.cs

```diff
@@ -0,0 +1,272 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics.Contracts;
+using System.Linq;
+using System.Web.Http;
+using Microsoft.OData.Edm;
+using Microsoft.OData.Edm.Csdl;
+using Microsoft.OData.Edm.Expressions;
+using Microsoft.OData.Edm.Library;
+using Microsoft.OData.Edm.Library.Annotations;
+using Microsoft.OData.Edm.Library.Expressions;
+using Microsoft.OData.Edm.Library.Values;
+
+namespace System.Web.OData.Builder
+{
+    /// <summary>
+    /// Contains extension methods for <see cref="EdmModel"/> to set the query capabilities vocabulary.
+    /// </summary>
+    public static class CapabilitiesVocabularyExtensionMethods
+    {
+        private static readonly IEnumerable<IEdmStructuralProperty> EmptyStructuralProperties = new Collection<IEdmStructuralProperty>();
+        private static readonly IEnumerable<IEdmNavigationProperty> EmptyNavigationProperties = new Collection<IEdmNavigationProperty>();
+
+        /// <summary>
+        /// Set Org.OData.Capabilities.V1.CountRestrictions to target.
+        /// </summary>
+        /// <param name="model">The model referenced to.</param>
+        /// <param name="target">The target entity set to set the inline annotation.</param>
+        /// <param name="isCountable">This entity set can be counted.</param>
+        /// <param name="nonCountableProperties">These collection properties do not allow /$count segments.</param>
+        /// <param name="nonCountableNavigationProperties">These navigation properties do not allow /$count segments.</param>
+        public static void SetCountRestrictionsAnnotation(this EdmModel model, IEdmEntitySet target, bool isCountable,
+            IEnumerable<IEdmProperty> nonCountableProperties,
+            IEnumerable<IEdmNavigationProperty> nonCountableNavigationProperties)
```

Comment:
consider pass IEnumerable<string> here, to resolve complex type case : 'Address/Zip'